### PR TITLE
Feature/refactoring

### PR DESCRIPTION
**Pull request by reneviering**
Hey,

thanks @goloroden  for your feedback. Based on it I tried to optimize the server part a little bit. 

The pull request includes the following:

- [x]  Indentation
Now we have 2 spaces consistent in all `*.js` files. I added `editorconfig` to keep it consistent by using different IDEs.
- [x]  Not all commands have authorization configured
I didn't configured authorization for the commands `hide` and `remove` because they should only be available for the owner himself. But you are right and I think it's confusing not defining authorization for them. So I configured them explicit with `{  forAuthenticated: false, forPublic: false }`.
- [x] Commands aren't using validation via middleware
Now every command has a basic validation with `wolkenkit-command-tools` by using `only.ifExists()`, `only.ifNotExists()` and `only.ifValidatedBy()`. To validate the process (for example a cache can only be found if it was published) I added some custome middleware functions. By the way I love the readability of the middleware validation concept, awesome! 😍.
- [x] `server/keys/auth0` is empty
I'm not sure if it's cool to have the `certificate.pem` inside the git repo. I decided to put it into the `.gitignore`. 

René

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -32,12 +32,10 @@
     "eject": "react-scripts eject"
   },
   "devDependencies": {
+    "eslint": "^3.19.0",
```
Comment (goloroden) [inline client/package.json]: I personally don't use the `^` in dependencies, as I have experienced problems with breaking changes multiple times (although the changes should be compatible). We _always_ use strict version pinning.

I don't mean that you *have* to change this, but I just wanted to let you know that this is a potential source of trouble.

File: package.json [hunk 2]
```diff
@@ -26,26 +26,19 @@
...
-        "docker": {
-          "machine": "wolkenkit"
-        }
       }
     }
   },
-  "license": "AGPL-3.0"
+  "license": "AGPL-3.0",
+  "devDependencies": {
+    "eslint": "^3.19.0",
```
Comment (goloroden) [inline package.json]: Same here… I personally would pin the versions to strict numbers here.

File: server/readModel/lists/ownCaches.js [hunk 3]
```diff
@@ -1,72 +1,81 @@
...
 const when = {
-	'geocaching.cache.hidden': (caches, event, mark) => {
-		caches.add({
-			name: event.data.name,
-			description: event.data.description,
-			coordinate: event.data.coordinate
-		});
-		mark.asDone();
-	},
+  'geocaching.cache.hidden': (ownCaches, event, mark) => {
```
Comment (goloroden) [inline server/readModel/lists/ownCaches.js]: Is there a special reason why you use arrow functions here?

IMHO (but that's a matter of personal taste), using

`'geocaching.cache.hidden' (ownCaches, event, mark) {`

is more readable, as it contains less visual noise. But as said, that's up to you.

File: server/shared/middleware/onlyIfCacheHasBeenPublished.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const onlyIfCacheHasBeenPublished = function () {
```
Comment (goloroden) [inline server/shared/middleware/onlyIfCacheHasBeenPublished.js:3]: 👍 for writing your own middleware.

Achievement unlocked 😊

File: server/writeModel/geocaching/cache.js [hunk 5]
```diff
@@ -1,145 +1,200 @@
...
+  coordinate: { lat: 0, lon: 0 },
+  countFavorites: 0,
+  published: false,
+  countFindings: 0,
+  comments: [],
+  removed: false,
+
+  isAuthorized: {
+    commands: {
+      comment: { forAuthenticated: true, forPublic: false },
```
Comment (goloroden) [inline server/writeModel/geocaching/cache.js:19]: You have changed this to alphabetical order. I personally would prefer the logical order. But what I do like is that you added the commands and events that are only visible to the owner.

File: server/writeModel/geocaching/cache.js [hunk 6]
```diff
@@ -1,145 +1,200 @@
...
+        text: { type: 'string', minLength: 1 }
+      },
+      required: [ 'text' ]
+    }),
+    onlyIfCacheHasBeenPublished(),
+    (cache, command, mark) => {
+      cache.events.publish('commented', {
+        comments: [ ...cache.state.comments, {
+          text: command.data.text,
+          author: command.user,
```
Comment (goloroden) [inline server/writeModel/geocaching/cache.js]: Do you actually want to use the entire user object here, including the entire JWT token?

**Review by goloroden — COMMENTED** [review]
Hey @revrng 😊 

Thanks for asking me for a review! I have looked through your changes, and commented on a few things.

All in all, the code got much better 😊

File: client/package.json (hunk 1 above)
Comment (reneviering) [reply client/package.json]: Thanks, makes totally sense to me.

File: server/readModel/lists/ownCaches.js (hunk 3 above)
Comment (reneviering) [reply server/readModel/lists/ownCaches.js]: No, there isn't a special reason. You are right, the readability is much better so I'll change it 👍

File: server/writeModel/geocaching/cache.js (hunk 5 above)
Comment (reneviering) [reply server/writeModel/geocaching/cache.js:19]: Makes sense, especially relating to the results of the event storming session it's easier finding the code for the event/command in the geocaching process.

File: server/writeModel/geocaching/cache.js (hunk 6 above)
Comment (reneviering) [reply server/writeModel/geocaching/cache.js]: I think the user id and some profile infos are enough. 👍

**Comment by reneviering** [comment]
Hey @goloroden 😊,

thanks for your valuable feedback. I made some changes based on it and the code feels much better now 🎉 😊👍 :octocat:

**Comment by goloroden** [comment]
You're welcome 😊

**Comment by goloroden** [comment]
PS: I suppose that you are going to merge the PR on your own, aren't you?

**Pull request merged**